dynamiqs/solvers/ode/adaptive_solver.py

```diff
@@ -14,18 +14,46 @@
 
 
 class AdaptiveSolver(AutogradSolver):
+    """Integrate an ODE of the form $dy / dt = f(y, t)$ in forward time with initial
+    condition $y(t_0)$ using an adaptive step-size integrator.
+
+    For details about the integration method, see Chapter II.4 of [1].
+
+    [1] Hairer et al., Solving Ordinary Differential Equations I (1993), Springer
+        Series in Computational Mathematics.
+    """
+
+    @property
+    @abstractmethod
+    def order(self) -> int:
+        pass
+
+    @property
+    @abstractmethod
```

Comment:
I think `tableau` should not be an abstract method of `AdaptiveSolver`. In principle, we should have `AdaptiveSolver` -> `RungeKutta` -> DormandPrince5` and then we could have `tableau` an abstract method of `RungeKutta`. For now we have a single adaptive method, and the next ones will also be Runge Kuttas with Butcher tableaus. But we could also have Adaptive Solvers without Butcher tableaus.